### PR TITLE
Post fixes

### PR DESCRIPTION
**Pull request by christinem**
This Pull Request just about completes the "Check My POSt!" page. 

This fixes issues #365 and #366.

The credit count now properly calculates how many credits are completed in each post - and the credit count turns green when a POSt is completed. The autofilling of textboxes for the 300+ level courses is now working correctly. Now, using cookies, the browser remembers the last active post tab that was open, and opens this tab when you switch back to the page. Also (a suggestion from David from our code sprint ) - you can now see the credit count of the last active post tab that was open in the main navbar, beside "Check My POSt!". You can see this both from the POSt page and the Graph page. 

Please feel free to try it out with different selections of courses, and if you find a bug somewhere, please let me know! I think I have covered all the edge cases but there is a chance I have missed a few. 

(Code is still kind of long and messy in this, refactoring will come in the next PR).

### CODE REVIEW

**Comment by christinem** [comment]
I think the merge conflicts are because the master branch was updated while I was working on this branch. May I merge master into this branch in order to resolve the conflicts?

**Comment by david-yz-liu** [comment]
@christinem Yes, indeed you should. Now that you're working directly off of master, you should keep your current working branch up to date.

File: js/post/update_counts.js [hunk 1]
```diff
@@ -1,8 +1,14 @@
...
+creditCountMin = 0;
+creditCount300and400 = {'Spec': 0, 'Maj': 0, 'Min': 0};
+
+
 /**
  * Updates number of completed courses in Specialist.
 **/
 function updateCompletedSpecCourses () {
-    'use-strict';
+    'use-strict'
```
Comment (Ian-Stewart-Binks) [inline js/post/update_counts.js]: Why did you remove the semi colons? From my note last week, this should also be `use strict;`.

File: js/post/update_counts.js [hunk 2]
```diff
@@ -87,14 +93,14 @@ function updateCompletedMinCourses() {
  * Updates number of 300 level category completed courses.
  **/
 function update300s() {
-    'use-strict';
+    'use-strict'
+
```
Comment (Ian-Stewart-Binks) [inline js/post/update_counts.js]: Too many blank lines.

File: js/graph/setup.js [hunk 3]
```diff
@@ -87,6 +87,7 @@ $(document).ready(function () {
 
     FCEPrerequisiteCourses = [CSC318, CSC454];
 
+
```
Comment (Ian-Stewart-Binks) [inline js/graph/setup.js]: Bad blank line.

File: js/post/update_post.js [hunk 4]
```diff
@@ -141,32 +124,36 @@ function updateAllCategories() {
...
-        
     }
 
+
     // Update 400s
 
     i = 0; 
     var spec400s = $('.lvl400spec');
-    for (var l = 0; l < 3; l++) {
+    for (var l = 0; l < 1; l++) {
```
Comment (Ian-Stewart-Binks) [inline js/post/update_post.js]: I could be misreading something, but this seems highly unusual.

File: js/post/update_post.js [hunk 5]
```diff
@@ -177,28 +164,42 @@ function updateAllCategories() {
...
     } if (countSpec < 4) {
         updateCategory($('#spec_extra')[0].getElementsByClassName('code')[0], 'not fulfilled');
     } if (countMaj < 3) {
         updateCategory($('#maj_extra')[0].getElementsByClassName('code')[0], 'not fulfilled');
+    } if (countMin < 3) {
+        updateCategory($('#min_misc')[0].getElementsByClassName('code')[0], 'not fulfilled');
     }
 
+    checkPostCompleted();
+
```
Comment (Ian-Stewart-Binks) [inline js/post/update_post.js]: Blank line.

File: js/post/update_post.js [hunk 6]
```diff
@@ -349,41 +385,64 @@ function fillExtra() {
...
             if (maj_extra[k].value.indexOf('MAT') === -1 && maj_extra[k].value.indexOf('STA') === -1) {
                 maj_extra[k].value = '';
+                maj_extra[k].readOnly = false;
+            }
+        } 
+
+        // add credit count for MAT and STA courses
+        if (spec_extra[k].value.indexOf('MAT') > -1 || spec_extra[k].value.indexOf('STA') > -1) {
+            creditCount300and400['Spec'] += 0.5;
+        } if (k < 3) {
```
Comment (Ian-Stewart-Binks) [inline js/post/update_post.js]: This also seems highly unusual.

File: js/post/update_post.js [hunk 7]
```diff
@@ -428,4 +493,129 @@ function fillMisc() {
 
 }
```
Comment (Ian-Stewart-Binks) [inline js/post/update_post.js]: Add blank line.

File: js/post/update_post.js [hunk 8]
```diff
@@ -428,4 +493,129 @@ function fillMisc() {
...
+}
+
+/**
+ * Autofills extra 200-level courses for last Minor constraint.
+ * @param {number} index The textbox number we are at
+ * @param HTMLElement} min300s Array of textbox elements to fill
+**/
+function addExtraMinCourses(index, min300s) {
+    for (var m = 0; m < 3; m++) {
+        if(index === 3) {
```
Comment (Ian-Stewart-Binks) [inline js/post/update_post.js]: Space.

Comment (Ian-Stewart-Binks) [inline js/post/update_post.js]: Why keep this in the for loop?

File: js/post/update_post.js [hunk 9]
```diff
@@ -428,4 +493,129 @@ function fillMisc() {
...
+            break;      
+        } if (getCookie(additional_min_200s[m]) === 'active') {
```
Comment (Ian-Stewart-Binks) [inline js/post/update_post.js]: Is this an else if.. or just an if?

File: js/post/update_post.js [hunk 10]
```diff
@@ -428,4 +493,129 @@ function fillMisc() {
...
+/**
+ * Updates the Nav Bar on the Check My Post! page
+**/
+function updateNavPost() {
+    var nav_post = $('#nav-links')[0].getElementsByTagName('li')[3].getElementsByTagName('a')[0];
+
+    if (getCookie('specialist') === 'active') {
+        nav_post.innerHTML = 'Check My POSt! (' + (specCount).toFixed(1) + '/12.0)';
+        setCookie('activecount', (specCount).toFixed(1));
+    } if (getCookie('major') === 'active') {
```
Comment (Ian-Stewart-Binks) [inline js/post/update_post.js]: I haven't seen this style of if statements before. I don't know if it exists. In any case, please use style that is consistent with the rest of the project.

File: js/post/update_post.js (hunk 6 above)
Comment (christinem) [inline js/post/update_post.js]: This was a bit of a cheat. Technically this part should be in another function. This is something I plan to deal with when refactoring.

File: js/post/update_post.js (hunk 10 above)
Comment (christinem) [inline js/post/update_post.js]: Understood. This is something I plan to do when refactoring.

File: js/post/update_post.js (hunk 8 above)
Comment (christinem) [inline js/post/update_post.js]: If you mean the `if(index === 3)`, that is a bound check to make sure we don't check for textboxes past the number of textboxes we have available to fill. The for loop assumes that none of the textboxes might be filled, so it loops from 0 to 2. But the index stops it early if some of the textboxes had been filled previously. 

(I should add more documentation for this).

File: js/post/update_counts.js (hunk 1 above)
Comment (christinem) [inline js/post/update_counts.js]: Ah right, thanks @Ian-Stewart-Binks. 
Just a question though, are you sure it's not "use strict"; (with the quotations)?

Every source I've seen seems to use this. Example:
http://www.w3schools.com/js/js_strict.asp
https://msdn.microsoft.com/en-us/library/ie/br230269%28v=vs.94%29.aspx

Comment (Ian-Stewart-Binks) [inline js/post/update_counts.js]: Oh yeah, it definitely is- I didn't mean to imply that it didn't! However, I can see now how it can be read that way- sorry! Thanks for the references, and thanks for asking for clarification!

Comment (Ian-Stewart-Binks) [inline js/post/update_counts.js]: `'`, of course.

**Comment by Ian-Stewart-Binks** [comment]
@christinem There are some redundancies in the Haskell code (`do`, `$`). I've started using a linter which is great at catching these sorts of things, `hlint`. You should be able to install it on a mac.

Are you able to go through the PostResponse file and fix these things? This is a short order, so don't feel pressured to do this tonight.

**Comment by christinem** [comment]
@Ian-Stewart-Binks can do, but I think I might do them in my Refactoring PR - just to lump all the refactoring and simplifying code together in one place. I'm working on that right now (once I fix the other little things here). Thanks though, that tool should prove useful!

File: js/post/update_counts.js (hunk 1 above)
Comment (christinem) [inline js/post/update_counts.js]: Not a problem! I was just making sure :) Thanks!!

**Comment by david-yz-liu** [comment]
@christinem At the very least, change the styling of the `if`s that Ian pointed out now, please.

File: js/post/update_counts.js (hunk 1 above)
Comment (christinem) [inline js/post/update_counts.js]: Done!

File: js/post/update_counts.js (hunk 2 above)
Comment (christinem) [inline js/post/update_counts.js]: Done.

File: js/graph/setup.js (hunk 3 above)
Comment (christinem) [inline js/graph/setup.js]: Done.

File: js/post/update_post.js (hunk 5 above)
Comment (christinem) [inline js/post/update_post.js]: Done.

**Comment by christinem** [comment]
@david-yz-liu ofcourse. I have mades the fixes for the `if`s.

File: js/post/update_post.js (hunk 4 above)
Comment (christinem) [inline js/post/update_post.js]: You're right, this was an oversight of mine. Thanks! I've fixed it.

File: js/post/update_post.js (hunk 6 above)
Comment (david-yz-liu) [inline js/post/update_post.js]: Okay, that's fine, but should this also be an `else if`?
